### ISSUE TITLE
[Feature Request] Support nuxt prop for Vuetify 3 on Routable components

### ISSUE THREAD

**Issue by oemer-aran** [vuetifyjs/vuetify#17490]
### Problem to solve
In Vuetify2 adding the `nuxt` to a routable component (e.g. `v-btn` or `v-card`) produced a `nuxt-link` if `to` prop or `href` prop was passed. In Vuetify 3 the `nuxt` prop does no longer exist.

### Proposed solution
This is the solution i would like to propose:
Without `nuxt` prop:
`<v-btn href="https://vuetifyjs.com"/>` => `<a href="https://vuetifyjs.com"/>`
`<v-btn to="/vuetify"/>` => `<router-link to="https://vuetifyjs.com"/>`
`<v-btn />` => `<v-btn />` (No `href` or `to`)

With `nuxt` prop:
`<v-btn nuxt href="https://vuetifyjs.com"/>` => `<nuxt-link href="https://vuetifyjs.com"/>`
`<v-btn nuxt to="/vuetify"/>` => `<nuxt-link to="https://vuetifyjs.com"/>`
`<v-btn nuxt />` => `<v-btn />`  (No `href` or `to`)

NOTE: `nuxt-link` can now handle internal (`to`) and external (`href`) links:  https://nuxt.com/docs/api/components/nuxt-link

**Comment by lioneaglesolutions** [comment]
@johnleider plans for this?

**Comment by johnleider** [comment]
Was hoping for a community contribution since none of the team uses nuxt.

**Comment by KaelWD** [comment]
Nuxt would have to expose a vue-router compatible useLink function on the component.

**Comment by so1ve** [comment]
Hi 👋 @KaelWD, I have raised an issue in nuxt repo: nuxt/nuxt#22169. Will vuetify team implement this feature once `useLink` function is added to `NuxtLink`? If not, how to implement this feature?

**Comment by LittleSheep2Code** [comment]
I noticed the button and others components use render function like way to render, we can use custom tag property to implement this feature, but will get a typescript error because the tag property doesn't accept vue component type.

https://github.com/vuetifyjs/vuetify/blob/a997c424f8660894555804bfcb88761cfead9238/packages/vuetify/src/components/VBtn/VBtn.tsx#L157C1-L168C54

The code will like this:

```vue
<v-btn :tag="RouterLink" to="/">Link</v-btn>
```

I didn't find way to directly use the NuxtLink component in code, so I made a wrapper of NuxtLink. That is the RouterLink referenced in previous code.

```vue
<template>
  <nuxt-link v-bind="props" />
</template>

<script setup lang="ts">
// Add other properties if you want
const props = defineProps<{ to: string; target: string }>();
</script>
```

I tested this solution, looks everything works fine. Except the warning.

<img width="500" alt="image" src="https://github.com/vuetifyjs/vuetify/assets/56875315/702f07b7-6932-43fa-9bd8-5fe79e072409">

**Comment by AnttechDev** [comment]
@KaelWD This has been done on Nuxt side: https://github.com/nuxt/nuxt/issues/22169, do you know when this might be expedited?

**Comment by KaelWD** [comment]
https://github.com/vuetifyjs/vuetify/pull/19514#issuecomment-2065784246

**Comment by alihardan** [comment]
> [#19514 (comment)](https://github.com/vuetifyjs/vuetify/pull/19514#issuecomment-2065784246)

@KaelWD, I agree with you. If it doesn't bring prefetching, it has almost no advantage. Maybe we should hope that prefetch will be added to useLink later.

**Comment by gsouf** [comment]
> Was hoping for a community contribution since none of the team uses nuxt.

@johnleider if no one is on I can tackle this one

**Comment by dpmillerau** [comment]
We are also affected by this issue

**Comment by johnleider** [comment]
> > Was hoping for a community contribution since none of the team uses nuxt.
> 
> [@johnleider](https://github.com/johnleider) if no one is on I can tackle this one

https://github.com/vuetifyjs/vuetify/pull/19514
